tests/stubtest_third_party.py

```diff
@@ -109,19 +112,35 @@ def run_stubtest(dist: Path) -> bool:
     return True
 
 
-def run_stubtest_for(metadata: dict[str, Any], dist: Path) -> bool:
-    return has_py3_stubs(dist) and metadata.get("stubtest", True)
-
-
 # Keep this in sync with mypy_test.py
 def has_py3_stubs(dist: Path) -> bool:
     return len(glob(f"{dist}/*.pyi")) > 0 or len(glob(f"{dist}/[!@]*/__init__.pyi")) > 0
 
 
+def install_apt_packages(dist: Path, metadata: dict[str, Any], install: bool) -> bool:
+    apt_packages = metadata.get("stubtest_apt_dependencies", [])
+    if not apt_packages:
+        return True
+    if not install:
+        print(f"Ensure the following packages are installed for {dist.name}: {', '.join(apt_packages)}", file=sys.stderr)
+        return True
+    try:
+        apt_cmd = ["apt", "install", *apt_packages]
+        subprocess.run(apt_cmd, check=True, capture_output=True)
+    except subprocess.CalledProcessError as e:
+        print(f"Failed to install APT packages for {dist.name}: {', '.join(apt_packages)}", file=sys.stderr)
+        print(e.stdout.decode(), file=sys.stderr)
+        print(e.stderr.decode(), file=sys.stderr)
+        return False
+    else:
+        return True
+
+
 def main() -> NoReturn:
     parser = argparse.ArgumentParser()
     parser.add_argument("--num-shards", type=int, default=1)
     parser.add_argument("--shard-index", type=int, default=0)
+    parser.add_argument("--install-apt-packages", type=bool, default=False)
```

Comment:
```suggestion
    parser.add_argument("--install-apt-packages", action="store_true")
```

Reply:
Currently using `--install-apt-packages` silently discards the first stub passed on command line. This is a gotcha with how argparse works. It might be possible to add an overload in typeshed to get a warning for this case. https://stackoverflow.com/a/15008806